### PR TITLE
codec: add metadata_not_supported_error to HTTP/1 codec stats

### PR DESCRIPTION
**Pull request by soya3129**
Description: introduce stats to HTTP/1 codec. Add metadata_not_supported_error.
Risk Level: Low. stats change only.
Testing: Unit test.
Docs Changes: N/A
Release Notes:

### CODE REVIEW

File: source/common/http/codec_client.cc [hunk 1]
```diff
@@ -140,7 +140,8 @@ CodecClientProd::CodecClientProd(Type type, Network::ClientConnectionPtr&& conne
     : CodecClient(type, std::move(connection), host, dispatcher) {
   switch (type) {
   case Type::HTTP1: {
-    codec_ = std::make_unique<Http1::ClientConnectionImpl>(*connection_, *this);
+    codec_ = std::make_unique<Http1::ClientConnectionImpl>(*connection_,
```
Comment (alyssawilk) [inline source/common/http/codec_client.cc]: Mind updating docs/root/configuration/http_conn_man/stats.rst ?

Each codec has the option of adding per-codec statistics. Currently only http2 has codec stats. needs updating, and a section on Http1 codec stats?

Comment (soya3129) [reply source/common/http/codec_client.cc]: Fixed. Thanks!

File: docs/root/configuration/http_conn_man/stats.rst [hunk 2]
```diff
@@ -98,7 +98,18 @@ following statistics:
...
+Http1 codec statistics
+~~~~~~~~~~~~~~~~~~~~~~
+
+All http1 statistics are rooted at *http1.*
+
+.. csv-table::
+   :header: Name, Type, Description
+   :widths: 1, 1, 2
+
+   metadata_not_supported_error, Counter, Total number of parsing metadata with http1 error
```
Comment (mattklein123) [inline docs/root/configuration/http_conn_man/stats.rst]: I don't think this description is quite right. Maybe something like "Total number of metadata frames dropped during HTTP/1 encoding" or something like that?

Comment (soya3129) [reply docs/root/configuration/http_conn_man/stats.rst]: Fixed. Thanks!

**Pull request merged**